### PR TITLE
Private/gokay/readonly mode

### PR DESCRIPTION
**Pull request by gokaysatir**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by gokaysatir** [comment]
Requires below changes:
https://gerrit.libreoffice.org/c/core/+/163430

File: wsd/ClientSession.cpp [hunk 1]
```diff
@@ -1120,7 +1120,7 @@ bool ClientSession::_handleInput(const char *buffer, int length)
         if (tokens.equals(0, "key"))
             _keyEvents++;
 
-        if (isEditable() && COOLProtocol::tokenIndicatesDocumentModification(tokens))
+        if (isEditable() && COOLProtocol::tokenIndicatesDocumentModification(tokens)) // check this function: tokenIndicatesDocumentModification
```
Comment (eszkadev) [inline wsd/ClientSession.cpp:1123]: forgotten todo? what is that comment about?

File: browser/src/map/handler/Map.Keyboard.js [hunk 2]
```diff
@@ -370,7 +370,7 @@ L.Map.Keyboard = L.Handler.extend({
 	// any 'beforeinput', 'keypress' and 'input' events that would add
 	// printable characters. Those are handled by TextInput.js.
 	_onKeyDown: function (ev) {
-		if (this._map.uiManager.isUIBlocked() || (this._map.isReadOnlyMode() && !this.readOnlyAllowedShortcuts(ev))
+		if (this._map.uiManager.isUIBlocked()
```
Comment (eszkadev) [inline browser/src/map/handler/Map.Keyboard.js:373]: We removed this read-only filter for keyboard. What do we have now?
User cannot open any dialog using keyboard shortcut for example?

Comment (gokaysatir) [reply browser/src/map/handler/Map.Keyboard.js:373]: Can you specify please? Keyboard events are filtered on the core side. We are planning to add some more checks but this should be working by now. If i missed anything, an example would work better for me.

File: wsd/ClientSession.cpp (hunk 1 above)
Comment (gokaysatir) [reply wsd/ClientSession.cpp:1123]: Yep, let me check why i did it, i will probably remove that comment :)

File: browser/src/map/handler/Map.Keyboard.js (hunk 2 above)
Comment (eszkadev) [reply browser/src/map/handler/Map.Keyboard.js:373]: no, just asking if we protect user from doing something by keyboard shortcuts. If it is filtered on core side - even better (it's always possible to emit something manually on the wire).

Comment (gokaysatir) [reply browser/src/map/handler/Map.Keyboard.js:373]: Ah okay thanks.

**Review by eszkadev — APPROVED** [review]
looks fine then

**Review by mmeeks — APPROVED** [review]
given that master doesn't compile without this and it's been through some review cycles; lets get it in =)

**Pull request merged**